src/wkwebview/mod.rs

```diff
@@ -1020,50 +496,31 @@ r#"Object.defineProperty(window, 'ipc', {
       #[cfg(target_os = "macos")]
       {
         if is_child {
-          let _: () = msg_send![ns_view, addSubview: webview];
+          ns_view.addSubview(&webview);
         } else {
-          let parent_view_cls = match ClassDecl::new("WryWebViewParent", class!(NSView)) {
-            Some(mut decl) => {
-              decl.add_method(
-                sel!(keyDown:),
-                key_down as extern "C" fn(&mut Object, Sel, id),
-              );
-
-              extern "C" fn key_down(_this: &mut Object, _sel: Sel, event: id) {
-                unsafe {
-                  let app = cocoa::appkit::NSApp();
-                  let menu: id = msg_send![app, mainMenu];
-                  let () = msg_send![menu, performKeyEquivalent: event];
-                }
-              }
-
-              decl.register()
-            }
-            None => class!(NSView),
-          };
-
-          let parent_view: id = msg_send![parent_view_cls, alloc];
-          let _: () = msg_send![parent_view, init];
-          parent_view.setAutoresizingMask_(NSViewHeightSizable | NSViewWidthSizable);
-          let _: () = msg_send![parent_view, addSubview: webview];
+          let parent_view = WryWebViewParent::new(mtm);
+          parent_view.setAutoresizingMask(
+            NSAutoresizingMaskOptions::NSViewHeightSizable
+              | NSAutoresizingMaskOptions::NSViewWidthSizable,
+          );
+          parent_view.addSubview(&webview.clone());
 
           // inject the webview into the window
-          let ns_window: id = msg_send![ns_view, window];
+          let ns_window = ns_view.window().unwrap();
           // Tell the webview receive keyboard events in the window.
           // See https://github.com/tauri-apps/wry/issues/739
-          let _: () = msg_send![ns_window, setContentView: parent_view];
-          let _: () = msg_send![ns_window, makeFirstResponder: webview];
+          ns_window.setContentView(Some(&parent_view));
+          ns_window.makeFirstResponder(Some(&webview));
         }
 
         // make sure the window is always on top when we create a new webview
-        let app_class = class!(NSApplication);
-        let app: id = msg_send![app_class, sharedApplication];
-        let _: () = msg_send![app, activateIgnoringOtherApps: YES];
+        let app = NSApplication::sharedApplication(mtm);
+        NSApplication::activate(&app);
```

Comment:
this change seems to be panicing, see https://github.com/tauri-apps/tauri/issues/11340

cc @lucasfernog @pewsheen @madsmtm

Reply:
https://developer.apple.com/documentation/appkit/nsapplication/activationoptions/1525559-activateignoringotherapps

https://developer.apple.com/documentation/appkit/nsapplication/4168336-activate

It seems `Activate` was added after 14.0, and `activateignoringotherapps` is deprecated after 14.0

Reply:
nice finding @pewsheen, so I guess we need to gate that behind a version check

Reply:
can't test but here is the PR, https://github.com/tauri-apps/wry/pull/1389 thanks @pewsheen